packages/react/src/views/Message/MessageToolbox.js

```diff
@@ -67,6 +69,7 @@ export const MessageToolbox = ({
     setShowDeleteModal(false);
   };
 
+  const isAllowedToPin = userRoles.some((role) => pinRoles.has(role));
```

Comment:
@devanshkansagra 

At this point, we are comparing two different rules. There is a possibility that one of the rules is in a different case.

It's just a possibility, but I wanted to check.
I'm not sure if that's the issue.

The problem might lie in fetching the wrong PIN permission list,
or it could be that the fetch operation somehow fails.

Reply:
Okay so you mean, the roles in the roles array?

Reply:
Suppose in userRole, my role is Admin 
In roles array, in pin permission, the role is admin 


Now Admin != admin
Hence check fails

Reply:
Hey @Spiral-Memory when I was researching for the solution of this issue, I somehow fetched the entire roles array from the permissions response and in that I have seen all the roles were having same case all small

Reply:
Ohh ok then, 
Try figuring out the issue and let me know

Reply:
Okay

Reply:
@devanshkansagra @Spiral-Memory 
In my POV , the code seems to work well for all user groups including admin , the reason why it might not have worked is that when pin roles are changed by admin in the panel, changes are instantaneously not updated in the user's permission store . Thus a user needs to logout and re-login in order to see changes in permissions

Reply:
rather than doing logout and login again is it possible to view changes on refreshing the page?

Reply:
i can try fixing it !!